framework/src/timesteppers/CSVTimeSequenceStepper.C

```diff
@@ -0,0 +1,69 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "CSVTimeSequenceStepper.h"
+#include "DelimitedFileReader.h"
+
+template <>
+InputParameters
+validParams<CSVTimeSequenceStepper>()
+{
+  InputParameters params = validParams<TimeSequenceStepperBase>();
+  params.addRequiredParam<FileName>("file_name",
+                                    "name of the file in which the time sequence is read");
+  params.addParam<bool>(
+      "header", true, "indicates whether the file contains a header with the column names");
+  params.addParam<std::string>("delimiter", ",", "delimiter used to parse the file");
+  params.addParam<std::string>(
+      "column_name", "time", "name of the column which contains the time sequence");
+  params.addRangeCheckedParam<int>(
```

Comment:
Rather than check the range of an int, why not just use `unsigned int`? That way the complier doesn't complain to us all over the place about comparing an unsigned int to a signed int. Yes, I see that you are using `-1` as a special value, but you could just as easily use the largest unsigned int as well for your special case.

Reply:
Changed. That should take care of the compiler warning below